pkg/apis/pipeline/v1alpha1/cluster_resource_test.go

```diff
@@ -0,0 +1,178 @@
+				Type: PipelineResourceTypeCluster,
+				Params: []Param{{
+					Name:  "url",
+					Value: "http://10.10.10.10",
+				}, {
+					Name:  "cadata",
+					Value: "bXktY2x1c3Rlci1jZXJ0Cg",
+				}, {
+					Name:  "token",
+					Value: "my-token",
+				},
+				},
+			},
+		},
+		want: &ClusterResource{
+			Name:   "test-cluster-resource",
+			Type:   PipelineResourceTypeCluster,
+			URL:    "http://10.10.10.10",
+			CAData: []byte("my-cluster-cert"),
+			Token:  "my-token",
+		},
+	}, {
+		desc: "no token",
+		resource: &PipelineResource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-cluster-resource",
+				Namespace: "foo",
+			},
+			Spec: PipelineResourceSpec{
+				Type: PipelineResourceTypeCluster,
+				Params: []Param{{
+					Name:  "url",
+					Value: "http://10.10.10.10",
+				}, {
+					Name:  "cadata",
+					Value: "bXktY2x1c3Rlci1jZXJ0Cg",
+				}, {
+					Name:  "username",
+					Value: "user",
+				}, {
+					Name:  "password",
+					Value: "pass",
+				},
+				},
+			},
+		},
+		want: &ClusterResource{
+			Name:     "test-cluster-resource",
+			Type:     PipelineResourceTypeCluster,
+			URL:      "http://10.10.10.10",
+			CAData:   []byte("my-cluster-cert"),
+			Username: "user",
+			Password: "pass",
+		},
+	}, {
+		desc: "no cert",
+		resource: &PipelineResource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-cluster-resource",
+				Namespace: "foo",
+			},
+			Spec: PipelineResourceSpec{
+				Type: PipelineResourceTypeCluster,
+				Params: []Param{{
+					Name:  "url",
+					Value: "http://10.10.10.10",
+				}, {
+					Name:  "token",
+					Value: "my-token",
+				},
+				},
+			},
+		},
+		want: &ClusterResource{
+			Name:     "test-cluster-resource",
+			Type:     PipelineResourceTypeCluster,
+			URL:      "http://10.10.10.10",
+			Token:    "my-token",
+			Insecure: true,
+		},
+	}} {
+		t.Run(c.desc, func(t *testing.T) {
+			got, err := NewClusterResource(c.resource)
+			if err != nil {
+				t.Errorf("Test: %q; TestNewClusterResource() error = %v", c.desc, err)
+			}
+			if d := cmp.Diff(got, c.want); d != "" {
+				t.Errorf("Diff:\n%s", d)
+			}
+		})
+	}
+}
+
+func TestGetClusterConfig(t *testing.T) {
+	for _, c := range []struct {
+		desc     string
+		resource *PipelineResource
+		want     *rest.Config
+	}{{
+		desc: "basic resource",
+		resource: &PipelineResource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-cluster-resource",
+				Namespace: "foo",
+			},
+			Spec: PipelineResourceSpec{
+				Type: PipelineResourceTypeCluster,
+				Params: []Param{{
+					Name:  "url",
+					Value: "http://10.10.10.10",
+				}, {
+					Name:  "cadata",
+					Value: "bXktY2x1c3Rlci1jZXJ0Cg",
+				}, {
+					Name:  "token",
+					Value: "my-token",
+				},
+				},
+			},
+		},
+		want: &rest.Config{
+			Host: "http://10.10.10.10",
+			TLSClientConfig: rest.TLSClientConfig{
+				Insecure: false,
+				CAData:   []byte("my-cluster-cert"),
+			},
+			BearerToken: "my-token",
+		},
+	}} {
```

Comment:
im a bit confused about how this test is different from the previous?

Reply:
cleaned up tests and added better descriptions

Reply:
im still not quite seeing why `TestSecrets` is a separate test instead of a test case in `TestNewClusterResource` ?

Reply:
combined them